examples/md-flexible/tests/domainDecomposition/RegularGridDecompositionTest.cpp

```diff
@@ -191,26 +196,49 @@ TEST_F(RegularGridDecompositionTest, testExchangeHaloParticles) {
   EXPECT_EQ(autoPasContainer->getNumberOfParticles(autopas::IteratorBehavior::owned), 27)
       << "Owned particles missing after halo exchange!";
 
-  // expect particles to be all around the box since every particle creates multiple halo particles.
-  const auto expectedHaloParticlePositions = generatePositionsOutsideDomain(*autoPasContainer);
-  ASSERT_EQ(expectedHaloParticlePositions.size(), 98) << "Expectation setup faulty!";
+  if (boundaryType == options::BoundaryTypeOption::periodic) {
+    // expect particles to be all around the box since every particle creates multiple halo particles.
+    const auto expectedHaloParticlePositions = generatePositionsOutsideDomain(*autoPasContainer);
+    ASSERT_EQ(expectedHaloParticlePositions.size(), 98) << "Expectation setup faulty!";
 
-  EXPECT_EQ(autoPasContainer->getNumberOfParticles(autopas::IteratorBehavior::halo),
-            expectedHaloParticlePositions.size());
+    EXPECT_EQ(autoPasContainer->getNumberOfParticles(autopas::IteratorBehavior::halo),
+              expectedHaloParticlePositions.size());
+
+    for (auto particleIter = autoPasContainer->begin(autopas::IteratorBehavior::halo); particleIter.isValid();
+         ++particleIter) {
+      EXPECT_THAT(expectedHaloParticlePositions, ::testing::Contains(particleIter->getR()));
+    }
+  } else {
+    const auto globalMin = domainDecomposition->getGlobalBoxMin();
+    const auto globalMax = domainDecomposition->getGlobalBoxMax();
 
-  for (auto particleIter = autoPasContainer->begin(autopas::IteratorBehavior::halo); particleIter.isValid();
-       ++particleIter) {
-    EXPECT_THAT(expectedHaloParticlePositions, ::testing::Contains(particleIter->getR()));
+    const int numberOfProcesses = []() {
+      int result;
+      autopas::AutoPas_MPI_Comm_size(AUTOPAS_MPI_COMM_WORLD, &result);
+      return result;
+    }();
+
+    // Expect halos only between domains, not at global boundaries, when they are not periodic.
+    const auto haloNumber =
+        (localBoxMin[0] == globalMin[0] or localBoxMax[0] == globalMax[0]) ? (numberOfProcesses == 1 ? 0 : 9) : 18;
+
+    EXPECT_EQ(autoPasContainer->getNumberOfParticles(autopas::IteratorBehavior::halo), haloNumber);
   }
 }
 
+INSTANTIATE_TEST_SUITE_P(TestHaloParticles, RegularGridDecompositionTest,
+                         testing::Values(options::BoundaryTypeOption::periodic, options::BoundaryTypeOption::reflective,
+                                         options::BoundaryTypeOption::none)
+                         //    ,RegularGridDecompositionTest::PrintToStringParamName());
```

Comment:
- [ ] Either implement the to string method or remove the commented-out code

Reply:
Removed, I think the output is clear enough. Just a copy-paste leftover ;D